### PR TITLE
Replace the test function with TestRunner

### PR DESCRIPTION
**Pull request by Cadair**
any reason why this wouldn't work?

### CODE REVIEW

**Comment by astrofrog** [comment]
Just to be clear, this requires your PR in astropy core to go through, correct?

**Comment by Cadair** [comment]
No, this works with current master.

**Comment by Cadair** [comment]
https://github.com/astropy/astropy/blob/master/astropy/tests/runner.py#L257

File: packagename/_astropy_init.py [hunk 1]
```diff
@@ -22,93 +22,9 @@
...
-
-    """
-    test_runner = _get_test_runner()
-    return test_runner.run_tests(
-        package=package, test_path=test_path, args=args,
-        plugins=plugins, verbose=verbose, pastebin=pastebin,
-        remote_data=remote_data, pep8=pep8, pdb=pdb,
-        coverage=coverage, open_files=open_files, **kwargs)
+import os
+from astropy.tests.runner import TestRunner
```
Comment (bsipocz) [inline packagename/_astropy_init.py]: I think think makes the egg builds fail as this stops ``packagename`` being importable without its upstream dependency. Would it be possible to put it back inside a function, like it was before?

**Comment by bsipocz** [comment]
@Cadair - Current master means that any affiliates using the template with this patch won't be able to run with astropy <v1.3?
That would be quite a drastic move I think, so it would be nice to we could come up with a workaround.

**Comment by Cadair** [comment]
No this was in master for at least 1.2, it is not something I have added.

On 6 December 2016 22:04:46 GMT+00:00, Brigitta Sipocz <notifications@github.com> wrote:
>@Cadair - Current master means that any affiliates using the template
>with this patch won't be able to run with astropy <v1.3?
>That would be quite a drastic move I think, so it would be nice to we
>could come up with a workaround.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/astropy/package-template/pull/211#issuecomment-265287578

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Cadair** [comment]
Might be better to stick it in a try block and dummy the function if the import fails.

Will take a swing at it.

On 6 December 2016 21:58:11 GMT+00:00, Brigitta Sipocz <notifications@github.com> wrote:
>bsipocz commented on this pull request.
>
>
>
>> -
>-    kwargs
>-        Any additional keywords passed into this function will be
>passed
>-        on to the astropy test runner.  This allows use of
>test-related
>-        functionality implemented in later versions of astropy without
>-        explicitly updating the package template.
>-
>-    """
>-    test_runner = _get_test_runner()
>-    return test_runner.run_tests(
>-        package=package, test_path=test_path, args=args,
>-        plugins=plugins, verbose=verbose, pastebin=pastebin,
>-        remote_data=remote_data, pep8=pep8, pdb=pdb,
>-        coverage=coverage, open_files=open_files, **kwargs)
>+import os
>+from astropy.tests.runner import TestRunner
>
>I think think makes the egg builds fail as this stops ``packagename``
>being importable without its upstream dependency. Would it be possible
>to put it back inside a function, like it was before?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/astropy/package-template/pull/211#pullrequestreview-11691320

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by bsipocz** [comment]
> No this was in master for at least 1.2, it is not something I have added.

Fair enough. We still have to come up something for LTS, or delay this a few more month, so 2.0 won't be that far away.

**Comment by Cadair** [comment]
So, this is something cookiecutter could help with, we could have a "support astropy LTS (1.0.x)" question.

On 6 December 2016 22:53:18 GMT+00:00, Brigitta Sipocz <notifications@github.com> wrote:
>> No this was in master for at least 1.2, it is not something I have
>added.
>
>Fair enough. We still have to come up something for LTS, or delay this
>a few more month, so 2.0 won't be that far away.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/astropy/package-template/pull/211#issuecomment-265299473

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by bsipocz** [comment]
Would that mean we need to have two parallel version of this? I have a fee concerns/questions for the cookiecutter, just haven't got to the point to writing them down, but stay tuned.

**Comment by Cadair** [comment]
we could just have a jinja template block if LTS then old way else new way kind of thing.

**Comment by astrofrog** [comment]
So what's the plan for this now the cookiecutter branch is merged?

**Comment by Cadair** [comment]
Drop it and start again I guess.
I can add a astropy LTS option in the template.

**Pull request closed without merging**